include/swift/AST/Module.h

```diff
@@ -744,7 +748,7 @@ class FileUnit : public DeclContext {
   // Only allow allocation of FileUnits using the allocator in ASTContext
   // or by doing a placement new.
   void *operator new(size_t Bytes, ASTContext &C,
-                     unsigned Alignment = alignof(FileUnit));
+                     unsigned Alignment = alignof(FileUnit&));
```

Comment:
Ah, that's a much more enlightening error.

Does the reference do the right thing *now?*

```c++
class Test {
  alignas(16) double x;
public:
  virtual void foo() = 0;
};

static_assert(alignof(Test&) == 16, "");
```

(Of course, maybe MSVC doesn't support overalignment, in which case the test won't work again.)